src/shell/dotnet/Shell/Shell.csproj

```diff
@@ -46,17 +46,37 @@
 		<ProjectReference Include="..\..\..\messaging\dotnet\src\Host\MorganStanley.ComposeUI.Messaging.Host.csproj" />
 	</ItemGroup>
 
-	<PropertyGroup>
-		<Fdc3BundleOutput>$(ComposeUIRepositoryRoot)/src/shell/js/composeui-fdc3/dist/fdc3-iife-bundle.js</Fdc3BundleOutput>
-	</PropertyGroup>
+	<Target Name="InstallMonoRepoDependencies"
+		Condition="!Exists('$(ComposeUIRepositoryRoot)\node_modules')">
+		<Message Importance="high" Text="Installing node_modules..." />
+		<Exec Command="cd $(ComposeUIRepositoryRoot) | npm install" />
+		<Message Importance="normal" Text="Finished command: npm install." />
+	</Target>
 
-	<ItemGroup>
-		<EmbeddedResource Include="$(Fdc3BundleOutput)">
-		</EmbeddedResource>
-	</ItemGroup>
+	<Target Name="RunFdc3LernaBuild">
+		<Message Importance="high" Text="Running npx lerna run build for composeui-fdc3..." />
+		<Exec Command="cd $(ComposeUIRepositoryRoot) | npx lerna run build --scope @morgan-stanley/composeui-fdc3" />
+		<Message Importance="normal" Text="Finished building the fdc3 bundle." />
+	</Target>
+
+	<Target Name="EmbedFdc3Bundle"
+			Condition="Exists('$(TargetDir)\fdc3-iife-bundle.js')">
+		<Message Importance="normal" Text="Embedding fdc3 resource" />
+		<ItemGroup>
+			<EmbeddedResource Include="$(TargetDir)\%(RecursiveDir)\fdc3-iife-bundle.js" />
+		</ItemGroup>
+	</Target>
 
-	<Target Name="PreBuild" BeforeTargets="PrebuildEvent" Condition="!Exists('$(Fdc3BundleOutput)')">
-		<Exec Command="cd $(ComposeUIRepositoryRoot)" />
-		<Exec Command="npx lerna run build --scope @morgan-stanley/composeui-fdc3" />
+	<Target
+		Name="EmbedComposeUIFdc3Bundle"
+		BeforeTargets="PrebuildEvent">
+		<CallTarget Targets="InstallMonoRepoDependencies" />
+		<CallTarget Targets="RunFdc3LernaBuild" />
+		<ItemGroup>
+			<Fdc3BundleOutput Include="$(ComposeUIRepositoryRoot)\src\shell\js\composeui-fdc3\dist\fdc3-iife-bundle.js" />
+		</ItemGroup>
+		<Copy SourceFiles="@(Fdc3BundleOutput)" DestinationFolder="$(TargetDir)\%(RecursiveDir)" />
```

Comment:
You don't need to copy anything, just use the `Fdc3BundleOutput` items as the `Include` attribute for `EmbeddedResource`.